### PR TITLE
feat(selectText): allow selecting all in the inputs and in the plain dom

### LINKED ISSUE #1537: [Feature] Easy mode for shortcuts
Right now pressing shortcuts can be a bit of a pain. Pressing multiple keys requires something like
```js
await page.keyboard.down('Control');
await page.keyboard.press('a');
await page.keyboard.up('Control');
```
If you want it to perform select all, you need extra code for Mac to work:
```js
await page.keyboard.down(isMac ? 'Meta' : 'Control');
await page.keyboard.press('a');
await page.keyboard.up(isMac ? 'Meta' : 'Control');
```

It would be nice to be able to do something like
```js
await page.keyboard.press('Control+a');
```
or
```js
await page.keyboard.command('selectAll');
```
@pavelfeldman

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/1537

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -1424,7 +1424,13 @@ The `format` options are:
 
 Focuses the element, and then uses [`keyboard.down`](#keyboarddownkey) and [`keyboard.up`](#keyboardupkey).
 
-`key` can specify the intended [keyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) value or a single character to generate the text for. A superset of the `key` values can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values). Note that only those present on a typical full computer keyboard are supported. Holding down `Shift` will type the text that corresponds to the `key` in the upper case.
+`key` can specify the intended [keyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) value or a single character to generate the text for. A superset of the `key` values can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values). Examples of the keys are:
+
+  `F1` - `F12`, `Digit0`- `Digit9`, `KeyA`- `KeyZ`, `Backquote`, `Minus`, `Equal`, `Backslash`, `Backspace`, `Tab`, `Delete`, `Escape`, `ArrowDown`, `End`, `Enter`, `Home`, `Insert`, `PageDown`, `PageUp`, `ArrayRoght`, `ArrowUp`, etc.
```
Comment (dgozman) [inline docs/api.md:1429]: typo: ArrowRoght

File: docs/api.md [hunk 2]
```diff
@@ -1424,7 +1424,13 @@ The `format` options are:
...
+
+Following modification shortcuts are also suported: `Shift`, `Control`, `Alt`, `Meta`, `ShiftLeft`.
```
Comment (dgozman) [inline docs/api.md:1431]: Should we support Opt/Cmd/Ctrl aliases?

File: src/dom.ts [hunk 3]
```diff
@@ -290,6 +290,12 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
     }, deadline, options, true);
   }
 
+  async selectAll(): Promise<void> {
```
Comment (dgozman) [inline src/dom.ts:293]: Maybe `selectText()`?

File: src/dom.ts [hunk 4]
```diff
@@ -290,6 +290,12 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
...
+    const errorOrNeedsInput = await this._evaluateInUtility(({ injected, node }) => injected.selectAll(node), {});
+    if (typeof errorOrNeedsInput === 'string')
```
Comment (dgozman) [inline src/dom.ts:295]: "needs input" part is confusing. Why don't we return error or undefined?

**Review by dgozman — APPROVED** [review]

File: docs/api.md (hunk 2 above)
Comment (pavelfeldman) [reply docs/api.md:1431]: It is unlikely that we see them in the tests though? They would be platform-specific.

File: docs/api.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/api.md]: Done

File: src/dom.ts (hunk 4 above)
Comment (pavelfeldman) [reply src/dom.ts:295]: I'd stay consistent with fill. Renamed.

**Pull request merged**